### PR TITLE
Add missing auto_pad attribute for ConvTranspose

### PR DESCRIPTION
**Pull request by axeldavy**
auto_pad affects how the layer is executed.
For tensorflow compatibility, it should be set to
SAME_UPPER.

Fixes: https://github.com/onnx/tensorflow-onnx/issues/765

### CODE REVIEW

**Comment by jignparm** [comment]
The unit test below is failing with the following message. Maybe the attribute value should only be set under certain conditions?

```
self = <test_backend.BackendTests testMethod=test_Conv2DBackpropInput_const_strided>

    @check_opset_min_version(10, "Conv2DBackpropInput")
    def test_Conv2DBackpropInput_const_strided(self):
        input_sizes_val = np.array([1, 10, 10, 3], dtype=np.int32)
        filter_val = np.random.randint(low=0, high=256, size=[3, 3, 3, 5]).astype(np.float32)
        out_backprop_val = np.random.randint(low=0, high=256, size=[1, 5, 5, 5]).astype(np.float32)
        _ = tf.nn.conv2d_backprop_input(input_sizes=input_sizes_val, filter=filter_val, out_backprop=out_backprop_val,
                                        strides=[1, 2, 2, 1], padding='SAME', name=_TFOUTPUT)
>       self._run_test_case([_OUTPUT], {})

tests\test_backend.py:2920: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests\test_backend.py:111: in _run_test_case
    return self.run_test_case(feed_dict, [], output_names_with_port, **kwargs)
tests\backend_test_base.py:131: in run_test_case
    self.assertAllClose(expected_val, actual_val, rtol=rtol, atol=atol)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

expected = array([[[[111087.,  95576., 113458.],
         [ 46745.,  99963., 104671.],
         [202821., 163341., 136903.],
    ...5466.,  55948.,  40606.],
         [ 86682., 150351., 198641.],
         [ 86307.,  59883.,  47439.]]]], dtype=float32)
actual = array([[[[ 93894.,  39163.,  34824.],
         [105692., 119105., 166645.],
         [ 58051.,  21856.,  26100.],
    ...9971., 152360., 152864.],
         [ 39489.,  94722.,  66264.],
         [ 89379.,  96516.,  85563.]]]], dtype=float32)
kwargs = {'atol': 1e-05, 'rtol': 1e-07}

```

**Comment by guschmue** [comment]
I'm not sure if auto_pad will work since we set the output_shape already. Maybe best would be to add a unit test for strides>1 and take a closer look if its numerical not the same.

**Comment by axeldavy** [comment]
It seems Keras uses SAME_LOWER for transpose convolutions:
[https://github.com/onnx/keras-onnx/blob/master/keras2onnx/ke2onnx/conv.py#L147](https://github.com/onnx/keras-onnx/blob/master/keras2onnx/ke2onnx/conv.py#L147)

According to this bug report:
[https://github.com/onnx/onnx/issues/1609](https://github.com/onnx/onnx/issues/1609)
There is a mismatch in the spec between SAME_LOWER and SAME_UPPER for this definition.
I don't know if that really can be fixed in the spec after so many months.

Thus possibly the correct auto_pad is SAME_LOWER. I'll check if #765 is also happy with this flag.

**Comment by axeldavy** [comment]
Testing with #765 the onnx fails to execute with SAME_LOWER under tensorflow (it indicates ConvTranspose with auto_pad SAME_LOWER" is not supported at the moment).
With tensorrt, the output is visually similar to with SAME_UPPER. I could do a numerical comparison to check which is the closest to the mathematical output of the original model if needed.

Additionnaly, even if this won't help the unit test, the flag should probably only be set for "padding == 'SAME'" (not setting the flag seems similar to 'VALID' according to onnx-tensorflow's onnx_tf/handlers/backend/conv_mixin.py).

My intuition is either the spec is considered wrong, and in that case SAME_UPPER would be the correct flag, or the spec should be followed, and in that case SAME_LOWER would be the correct flag, and in that case onnx-tensorflow is probably incorrect (mixing the two flags for tranpose convolutions).

**Comment by axeldavy** [comment]
Given the discussion above, I think my patch is quite incorrect (should only be activated for SAME padding, fails a unit test, probably is unrelated to stride > 1 and should affect stride = 1 too), and there are possible required changes outside of tensorflow-onnx (onnx spec, and/or onnx-tensorflow).

It seems to need a harder fix than I would have thought, and I think it is more appropriate that I leave this issue to a more experienced tensorflow/onnx developper.
